### PR TITLE
feat: [PLG-2533]: added feature flag for cd onboarding wizard with  harness-cli

### PR DESCRIPTION
**Pull request by devinder-harness**
## Describe your changes
added feature flag for cd onboarding wizard with  harness-cli

JIRA:
https://harness.atlassian.net/browse/PLG-2533
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49766)

### CODE REVIEW

**Review by neelam-harness — APPROVED** [review]

**Review by kapil-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49766)* status: :shipit: complete! ✓

**Comment by devinder-harness** [comment]
trigger messagecheck

**Review by Xin-Harness — DISMISSED** [review]

**Review by neelam-harness — DISMISSED** [review]

**Comment by devinder-harness** [comment]
trigger messagecheck

File: 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java [hunk 1]
```diff
@@ -807,6 +807,7 @@ public enum FeatureName {
   WINRM_SCRIPT_COMMAND_SPLIT_NG(
       "Enables the new way of how to copy powershell/winrm script commands content to file on remote. (Copy is done in chunks of 6KB) ",
       HarnessTeam.CDP),
+  CD_CLI_WIZARD_ENABLED("Enables new cd onboarding wizard with harness-cli", HarnessTeam.PLG),
```
Comment (arvind-choudhary-h) [inline 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java:810]: You can add `PLG_` as a prefix here. `PLG_CD_CLI_WIZARD_ENABLED`

**Review by arvind-choudhary-h — APPROVED** [review]

File: 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java (hunk 1 above)
Comment (neelam-harness) [reply 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java:810]: Good point. Please also update the team name to `HarnessTeam.GTM` and add `GTM` as a prefix.

**Review by neelam-harness — DISMISSED** [review]

**Comment by devinder-harness** [comment]
trigger review

**Review by arvind-choudhary-h — DISMISSED** [review]

**Comment by devinder-harness** [comment]
trigger uts

**Review by abhinav-harness — APPROVED** [review]

**Comment by devinder-harness** [comment]
trigger review

**Review by neelam-harness — APPROVED** [review]

**Review by vtxorxwitty — APPROVED** [review]

**Comment by devinder-harness** [comment]
trigger review

**Review by inderschera — APPROVED** [review]

**Comment by devinder-harness** [comment]
trigger uts

**Pull request merged**